Revise F# formatting guide

dsyme opened the pull request:
## Summary

* [x] Rearrange sections of F# formatting guide to follow a logical order
* [x] Add some clarifications 
* [x] Add a comment saying blank lines should come before `///` documentation comments
* [x] Consistently use "multi-line" not "multiline"
* [x] Address https://github.com/dotnet/docs/issues/25917#issuecomment-913516706 
* [x] Removed the naming conventions from this doc.  F# naming conventions are in the F# component design guideline, they were inconsistent with the naming conventions here
* [x] There was a comment saying "put two blank lines between type definitions" - I've removed that and just said one - I don't think it really makes sense to differentiate that and in any case doesn't cover what to do between "let" and "type" definitions, nor "type" and "module" etc.
* [x] added some information about whether to put spaces after a function/method name in an application

## [Preview Link](https://review.docs.microsoft.com/en-us/dotnet/fsharp/style-guide/formatting?branch=pr-en-us-25972)

dsyme committed:
Revise F# formatting guide

dsyme commented:
@nojaf Could you review?  

This addresses https://github.com/dotnet/docs/issues/25917#issuecomment-913516706 but also rearranges the sections in the guide to follow a logical order.

Please check these in particular

* Recommend to always ensure a blank line before any `///` doc comment. Always - including for members, union cases, record fields

* I've left the record declaration formatting guidance in place but noted that if record fields have `///` comments then the formatting used should expand to accomodate this, including blank lines and the initial `{` on a separate line.  I think this seems right but we should discuss and ensure Fantomas aligns

* I added an explicit link to Fantomas and recommendation that a code formatter should be used.

dsyme committed:
Update formatting.md

dsyme committed:
Update formatting.md

dsyme committed:
Update formatting.md

dsyme committed:
Update formatting.md

nojaf commented on docs/fsharp/style-guide/formatting.md:
````diff
@@ -8,103 +8,50 @@ ms.date: 08/31/2020
-let myFunBad (a:decimal)(b)c = a + b + c
-```
+We strongly recommend the use of this code formatter. Within F# teams, code formatting specifications should be agreed and codified in terms
+of a agreed settings for the code formatter. If the code formatter doesn't implement the setting you require, contribute a new
````
Could you please leave out the sentence `If the code formatter doesn't implement the setting you require, contribute a new setting to the project.`. This is really not the case, our [contribution guidelines](https://github.com/fsprojects/fantomas/blob/master/CONTRIBUTING.md#what-are-we-not-looking-for) are very clear on that.

The impact of settings can be huge and we had problems in the past where people do a one-time contribution, introduce massive technical dept and are not there to clean up the mess. Each time an FCS update or editor tooling changes we need to deal with all those things. It is not worth having currently.
That is why we currently stick to two style guides and that is it.

Keep in mind that Fantomas at the time of writing cannot even fully format the F# compiler itself. That is a higher purpose to achieve than trying to keep up with everybody's little nitpicks on how stuff should look like.
It is not worth it and it burns people out. Once the compiler can be formatted, we might revisit our policy but until then we really have bigger fish to fry.

dsyme replied:
Got it, thanks, will do

nojaf commented on docs/fsharp/style-guide/formatting.md:
```diff
@@ -116,378 +63,362 @@ This is sometimes called “vanity alignment” or “vanity indentation”. The
 
-It is common and accepted F# style to use camelCase for all names bound as local variables or in pattern matches and function definitions.
+// OK
+let myRegexMatch =
```
Should this not be something like
```fsharp
let myRegexMatch =
    Regex.Match(
        "my longer input string with some interesting content in it",
        "myRegexPattern"
    )
```
Seems a bit inconsistent otherwise.

dsyme replied:
I'll change it for now.  However I don't think the guide is sufficiently clear on parenthesis location - for application of DUs it says either is ok, but this section implies
```fsharp
let myRegexMatch =
    Regex.Match(
        "my longer input string with some interesting content in it",
        "myRegexPattern"
    )
```
is normal.  What are the default fantomas settings?

dsyme replied:
I've checked the fantomas default settings and documented them in the style guide.

nojaf commented on docs/fsharp/style-guide/formatting.md:
````diff
@@ -962,26 +745,42 @@ match l with
 ```
 
-Pattern matching of anonymous functions, starting by `function`, should generally not indent too far. For example, indenting one scope as follows is fine:
+Pattern matching via anonymous functions, useing the keyword `function`, should indent one level from the start of the previous line:
````
`useing` typo?

nojaf commented:
This was a bit hard to review, so I may have missed some things.
I think the reordering is fine.
The remark to recommend Fantomas is much appreciated 🎉!

As for the blank lines above comments, if the original code already had them Fantomas will do the right thing.
If they are not there, I'm not sure if it will be applied in all scenarios.

dsyme commented:
@nojaf Other changes

* [x] Removed the naming conventions from this doc.  F# naming conventions are in the F# component design guideline, they were inconsistent with the naming conventions here
* [x] There was a comment saying "put two blank lines between type definitions" - I've removed that and just said one - I don't think it really makes sense to differentiate that and in any case doesn't cover what to do between "let" and "type" definitions, nor "type" and "module" etc.

dsyme commented:
@nojaf I'm going through the entire revised doc using the Fantomas preview online tools to check the formatting.

Could you check this one?  I was expecting
```fsharp
Target.create "Build" (fun ctx ->
    // code
    // here
    ())
```

https://fsprojects.github.io/fantomas-tools/#/fantomas/preview?data=N4KABGBEDGD2AmBTSAuKAVAhgJwOaIBcA6abRTAxMAHUgCEBXASwBt5awAKAMwYDsw0AgA8wAWgB81bNQEQwAegWCEiabPmLlAC0Rl1ciJwCUxyABpwUAM6ECTPrmuowAbSsRQmiJAAkBAE8AB2Q0SAcCCw95PwA3TBYGULdozQAGS28YgEk%2BJD4CAGUmAC9kTKywABZUsABdaIBfCs9av0CQl3CCqKy4hKSXd0qwAEYW70gAWUxhABkHRDnERwJtXsrRgCY02obNZuivSf9g5MgAI1hYFg3NfsTk4cqtifvCxABbJjgWWD4AIIEACieQA8twFnxyrUINwErY9k0Jsd7qdOmErjc7jFfPFHkNYWAAMxvGKFIKYaCIOiIbiwMgABRwmE%2BhD0OM0BGwSSRBxRbXR5yxtzJD0GaGeWSqZJ8FKpNLpDKWsAA7npoJhbLlYrBNfZ-pz5NzeVl9vJDppUbiOsLrqLBfiJSkRgBWWVQeXU2n0sgAVSCIWwmu1fF1%2BqYho98JYiLNyKOgttXRFRvFTyJADYPZAvYrfYgAMIsLXWQv-awmoQMo1whGIPkWgV9IUp%2B1pvEDDMjADsObzPuVUy%2BFw50frjYglvk1p8rcx7bFnYJkqJAA5%2B5TvUqyOW-nxa2AY3HvOap82TsmF9il07u5UAJybhUA7iUbDlz6fTCHk0N%2BP8omLZXlAqa3l2hIjKMGRErmW75sqHzfL8UZEse-6ngmVpJmcbY3o6EGrlB4ywXmr7vkhPw3KhIx-pOYDTq0wG4deDotnekGbK8sG5PkBBgnw6DYAEADqTBrIe6H0YxYCzlA86gYuBEri6mykqR8EAtgsD8PAAAiiAsEw3zvr%2BPIYZoZ4MReaIgd0kTgSpUreKMMqwTMwjZNw6C6HwwKxoghTaAyBBifAEkelUuwAU2QGXixUARB2HFEZs7rubMuTcEwwhgkGFAMsCwhBGQ1jWJGB4eq60WYYB2HMRiiU9I5zrOZoozZhlwgAEqIHA2DwGFEVElF0k2TaCX2clhGqVkox9l1vX9fAAByDCfKO2AQtklCfM4HqjGNcW2ZNnwMCw9hGdCABiDLfgQ77oLhLX3nNG6wUtDLwFM52XYst3YPdpk5tA2gskIegAPqqkw4XrEd9XxY1U0vZxc1Pl1WnYJgARgtgCyVkN6yRTVllYTOOHI0lqOpVkOw5h5WM43jBMEGtG16Ntu37USh0xeex0TcjZ0Xaw-13RQj3PcprVElsJEjJATO4-jTCVj9otXYgANA2OsGg%2BD77Q7DEkIxTDXnNTMuvd4WzcYrHkAGpdnQDjwA4uBE4eo389ZgtznZVvsTNbXyFs6kO7M138EIFWu3kHteyTZtMUjlvNdbaO225kfCHpsAEAA4oQRUlYgZUVUnRLVSnsmU3a%2BHB05cvpQ7v1i9CdB-NAADWdDY73hDWAJhSskWNzrZVaETr7Mlye0k1gZntO251isrYgqpa7SBDqisT0hAZ2V8OJFUAnkw4c9gPMjFJs-jQHi9KU3ssjFsC2KwA0ogiBBF5PkrFyKPNkUIYS3xnrVWKiMTpUwzi-G2mgtjvVzv5b41htBVxGD7SBAtoFCwbmxE4KVZq2wxorYojgWCIC0rgdaKxQqIAuFMVQkkIFkzqubNOeFCFomIaHCAxIYKKwBEZXAfBo58Fjv8YoYiKAMDIOgWAvF6EUAqqwgKtd54KUuM-IhIciTEgVpUJWF09B8FUbEFUjhL6bSPg4U%2BFZ1EnnYVAzhMCCHTWbiMYk9tjEa3sKAuYrILjwEwMWWA5VHAby3osJxFl5BWTnvXbhnjX6VGJBHYxek1aYAuFQ1BattCFACAUWYcTNHJLCCseAEJQGFECFQ1JCD5DEhzsY0ENTISxJBtgFg3BIAVItikmmJDNDElbsY7%2Bv9lEFFyP3TAkjibTw0fff28k7JL3gVnMZa9jFdwWT3UB1gtI6TyBvSsiBvrty1qXUq5VHEejoqsvBj9kabL0V49JH89k4EHGQbJ1hSDGQcBQS5foT7-AMtAEs2MDRT3ASsnBfsXnrKfo3D5aSsjEmQcY%2Bp2AfgEGYUgcps8rB1AqOEaw11youCko0IAA

dsyme commented:
@nojaf Also this one, which formats the `+` at the beginning of line not end of line - I guess that's by design?

https://fsprojects.github.io/fantomas-tools/#/fantomas/preview?data=N4KABGBEDGD2AmBTSAuKAbRAXMAzArgHbRYCWshAjGAIYBOA5tfQwEy2MDMHDALGAF4e1ANQ92Ylt0mN%2BMtmHnSec8Yp7KWvADp1thSABpwUAM7YyhBqdRgA2iYigILqABIsATwAOyNJFJCLCNHF0g3ADcadHw-e1DXAAZjV1dIAElCJCCAZVIAL2QU1IheBLAAXVCAX2KncvCvX1sAoJCS8KiYuIcSiEo61MgAWRoADwAZQMQJxCssAAt2vspWRPKq11rQ5yGPHzjIACNYWHRltMjo2NtevtZBtJzEAFtSOHQKAEEsAFEsgDyuCmhCK5QguGi5g2NUGu0uTUOJzOFzCV26t3BYE4jzCOW8NGgiAAQohcLA6IgAAr0GgvbCIOio1xYOixGFbOENfbNfzI864zrXHpY3i4iCQfGEklkikzWAAd0Z0Bo5kyEVgKrIFGZLlZ7JKmxc21c8LRiJa-N1QoxaDuJQArOKoFKiaTyZSAKreXx0FVqwgarXkAzOyHoaGG2E7bkWvmnAXcro3O1YgBszslBLdsspAGF0KrTHmKKZ9SQKbqIVDEBzjVyOjykQnreiU-E%2BmAAOyZ10yj2IYavI6Mqt4Gt1iAmlxmiVNy0twVtkWdgAcvez-blJc%2Boax4cjqSNU4bezjUCtS%2BTK76AE4N9KvrgsIySy8XjQx-ra1HOTHG%2BexyLkmwqYp2lDJFiWbSu6crPG8Hw6mGE6-vW-5ngcC4oleoGpuBAxQX2T4vnQ8HvGcSFYt%2Bk5gNO9QAZh8bYSBtodisDxQZk2RYAChAACp0J4ADqpCLGOB4-ke0amrGjEXsBjbXmBKw4oRm5fHQsBEPAAAiiDoKQbwkV%2BbKSa4x60aeCJya0wQ4ax9qpJQYpQaMYzpLgfELHMvwRogOQLBSWAifAYnOrw6yoSe6HWbyUCBHZLHto5riUE6rnjJkuCkGMAK%2BjQWAUr8YzeJSpimCGY4OpFUl-jJDFxbZrZKXhKwZhlYwAEqIHAdDwCFYWijV5nSTOsmNQlzW4WxJSUD2HXdb18AAHL4C8I50EC6Qvi8NjOpQNF0WAs7uIBLz4OgZAGaCABiFIflgJF8Zh9nJVilDrlBi0UvAwwXVd0x3XQD3GZm0ALLSJCMgA%2BgqpChUsh1WeaNmTa9N6zfeHUaXQNCeACdBTGWA1LOFw0uBZR0nY0qNtOjyklGsmZuTjeME0TWCretjJbTte3vUjMUo4152XaQ12IEDIOMs9zT061jMEZ2kCs-jhOkGWf1ixLUsFaDUHg5DJGw-DYmC-VGETXTSUY6krAccrbkAGrCsSgTwIEDAk2OEXm2NDWHGjNsM3bqmO%2BMN1ECQIZu1knve2Tfv0ZbgfW4p00pS4rAueHYw6bAWAAOLYMVpWIOVIYJ1i1VJ8d43Nsx6cOVirDpY7-3i9MxKfNAADWxK4332CmLxOR0ogO5rXunYSbX1PzkxiZN29nasO1yvLYgCoS6SWBKnMsuIHp2WEKJIZfFkQ7c3Q-MzyhtVoRbsUN0vewtTNdvzcrADSiCIN4HkvJzEyGPekIIwR3z8nPeuLQg7L1tq4Vgn1c6%2BTeKYBYVdOy%2ByipZIWc5AKXmDgrO2WNlZ5CsJgDSDA1pzGCogI4wwEAQL6LPHBVMYGLyms3TsnBILKy%2BAZBghBI7EG1IQPIQiCr4EpHxWAXFaEFUqshKBbDkb4JsoQ%2BBIdXCcCVn0FWl1GSEEUREeUVgr4bWPoEM%2BpZxL3xGnVf2KcsKv0uO-TOEBOAO30VrMg4CJh0iOPAGgBZYAVSsJvbe0w7EqIftFJ%2BwsX5cJXn0TgYd9E6Q1jQI4mBUEawWDkTwQRxgxMPA4x%2BTjn4tDmPAIE4CcheEwMkhBLhOA530f8WpwJolgzoOgXAkBoEBxcc07RrS276N-v-eRQRMgDxoMQUm%2B57EU1GsnKpnD5Yfx0evfR3cFm93AaYDSWksibzLIgX6HcJalzKhVWxzpqKqLwadDRCk34ZyxJwL%2Bez6CwUpJk0w0A6CGUCAVS5npT4UD0tAQsuMxGlLMqsxx6zEkjK2R47EyD9ENJBSQRhSBEWHRMBUYoARTA3QqrYWe1QgA

dsyme commented:
Maybe check this one too

https://fsprojects.github.io/fantomas-tools/#/fantomas/preview?data=N4KABGBEDGD2AmBTSAuKAbRAXMAnAhgJYB2ARrAO4BMYAvADq73EQTB5FmVgWFYAWjZqxFgAQrADOkumHqQAUogBmy3IgCe8oS1EQAMvmgBrTTNpgA2nMgAtQgAcHWyAG4bAcUSxcAc0Ty7vJiAK7EvpjyYAC6YAC%2BOpAANOBQkthYJL6SqFapbPmskAAkWBoOyGiQJFjJhRAlAG746CGVeXoQAAwpnVAAksRIxFgAyoQAXsi9nQAs9TGFcTMFeiVlFbnVI3Vrxc2t7ZYLEACMK6KQALL4AB76JIj6iOECu52nVF0L0UsXoHsNu1IORYOh3iImi02rljn0qBdIaNEABbQhwdCwYgAQSwAFEhgB5ZQPYjTE5gZQtdI-P6FAGXUrlYGg8GIqD7aFHCkAZnZDVGDiMiDEKh8iAACvgCCjsIhcBDRFhcG1aSJlvSFutmVtWYqGpzDrCKbN%2BVBBcLRcpxfpKPLoPh0oNGrAHZksfrWMrVXpfur-lqmZsqnr2VCjWg4Z0AKxmyAW6AisXqACqTntjsQztd%2BHdxE9ECp6BpvrpIgZkKDLNgYM94ZhkYpADY4wmk9b1ABhdCOySdrGSb3QLA%2BAuU6mINWsDXlwNA3U1tmBg4NjqdADsraFiat4quqNI8rHRZLoj904DgJ1IcXdcNq6jegAHFvLcnEP3MfmzSfJ6X-ZqV7BlAobLlyxp9AAnK%2BibYsoWDyv2KIovgY7en%2BZ5lqwFZFFWC61mG97cn0pw9BS8bbu24rImiGIej%2BE5ThAM7YXO14gbehErsRHznORbZwQhuA0eiYL0RS6FMfEl6MvON4EWBEZrnonxxoMwxYISxAACq4BoADqfD8MejH-hegGyexIKcYpD4UqcfL8ZR2K4LAYTwAAIog6CEGiQloSqGEiOezEyZWclQDUd7cRBHymuRNy3P0yjafwLx4sWiCjPwPhYIZ8BvGaszfGZoUWeFVlRVx4GNiRsYJXcgzKIQtyEhUBAjrgeK3A46jSIQ4l9NGJWYQBs5AcCVW2TxKktg1twAEqIHAuDwPlhUmiNwVYaslnAdstTVUpj6iKcm7zUtK3wAAciEKKHrgxL9AhKI5GapxSSxu0VftKIhOgmQ%2BWSABiPgoVgQnaTqR12SRL7kZdPjwFc-2A48oO4OD-lxtA-DSkYQkAPq8AVxmfWFuERQd0U1cpp3QfNLkEBohK4A8g7rcZRVbawIXSeVlOVTsMMzaIXxxolTP4CzbOEIOt33fKT0vW99nkwLBpU39AOEEDiAY1j8pQ5sIuxXoVB8X0kBSzL7NYCjOt6wbubY%2BRuP48O8rE4QpOQOr417ZNwvTWbYsIvNABqXJiCQ8BZJzY7Ff7rETVsU17DFtWdFQjlW4lwNhMOA3EDHQzxz7G19EnpX8wHP1B4dIdZ%2Bb8V53cHmwFgXj4j1fWSMXCdmsNyffYL%2B2gRntMnSIVD1XnqO648YiYiYYgECY2CSFpoz4LKn53d%2BFK-iPYA4ZrVkT4ymd0zPc1W9diAUHropYBQiAvMbiBec1xB8MX2JDH3IrXAqs%2BjHxrl9U%2BbFx42UnsdCkVBzpWwANLvwcMlVKLxBg71lKSckYDTKjXMnXMeDcabwPhPDNutwMpokkPwQem0T5nw5FTS%2BlZr7T1YFQBmVtxjhEwC5Xwd0Xh5UQKQK4CB8GdHAUQsqJDz4wIUnA2GnQeRkSttiHyvhiAF2IEXLE4wdG5hCOobSsB1KiNzMXEymVmHQOrMoq%2BU9eSW06NbAG8piDWMaE8LEvggEPW-iQP%2BA5bGnm2mNFOgd8JLhUaLEQPJw7zx1ngww914D4G7FILID8n6PHCUFXmO0oGp3knE5xFC1G53cR5OW%2BBSCYFoXLfgowNAjDuIU%2BxZSoAvHgMSPBowyiYHIaovQPJW7uIJP0kkBSca4HQMoP2ECKaKMcRUjhLi%2Bg8jnu41BiB0FDFEYMNe%2BB9FcyPoQyJxDon11iaMhJrAeR33ccvM5xg8GSBcm5IYD9ByIGRgvPW3VeqIH6mEs0kkVka1YRfWBlSxmiB5Eg150pdzqDqZIaAuBfIkFzAClMv8sReWgD2DqNiGJ2OhQo2FSiNm4U4byKh7ihk4uHJIpAXSIGpGiDMaokhgb91yMfOIQA

dsyme commented:
@nojaf There's also a discrepency with respect to named arguments - the style guide says "no space"

https://fsprojects.github.io/fantomas-tools/#/fantomas/preview?data=N4KABGBEDGD2AmBTSAuKAbRAXMBbAhgNaIDKWATovrgEpVLlgAeYAvGAHaIDuYJAngGcsiXADoAkgHkxZStTr4GACgAO%2BLAAtWTAJQAdcvo6QANOCiDsWAJYcA5oNRgA2hYigIXqABIs-VWQ0SDssM3cvSB8AN3x0AFcg1wjvAAZzb29ICQ4kDiwSGwAvZAzMiAAWFLAAXQiAXzKPaqj-QOcQ-PDyqNiEpLdyiABGJszIAFl8JgAZO0QZxActbqHhgCZU6rrvRojPcb8ApMgAI1hYdFWsmLjE50Gh9bGsklEbOHRYDgBBLABRXJSABmcy410ywLiVm2DTGBxubRO50uEIgvTuA2qEAAzC9IiR1NBEAAhRDA2CUAAK%2BHI1GwiHIaK8FESsN28JaR3awRRV3xGP6D2xYAq%2BPRhPwxLJFMoM1g3EZ0HwVhy0Vgyts32ZEFZiHZXj23gRkW5yIu-K5fXuaEe5QArOKoJLpeTKYgAKqqQLkZWqjjqzU2bVOqHoGHlHaGzk9M0dPnMwU25JDMAANidkBdpLdlAAwugVYI899hOR4tAsJSdWAwxHMlGIEavCb0XHeRbE7chbaRQB2TPZmXuiaiU6Mmt1-WRuH7LlI%2BOdgXd5N2zIADkHRJzssQJa%2BJlD0OnDdnxvnx0XqOX1qxqYAnFupYgfsCROQS7gCDW9QamzHDgXDtrytTFhVTYZ0hFLNt2HSg3lwD5LhDEUpz-MBm2aWMgKgBMbzA3sINGaDs1fd8EKQg8f3LE9vEbDCAMRS9gMtWNb3AtZnmgnI8iwKQOAAFXIfgAHUbBWI9w1orx6MwsBW18HCziXUCexTNY8RI7cfnIWB4lyAARRB0BsRD32otkZw5OdsOYqBQi7djCLWMVoKmJgJGBATNCWf4pJITRKSwMT4AkkUKi2KzoxswC7M6MJ8LUtdvGGR03OmHJgRsJgpB9DRKX%2BJhVEoQRBGDQ8RXtSLT2s89bJ5eyukS1cRWGDN0qYOg4HIeAQrC1MIvQuSFNaOKHOau81gHDquspeAADl4lwcdyBBCQRFwJwnWGIbGNNJTcHidBbBMrgADFKQILB3wEy8Jo48phk3aDZp6iYjpO%2BYLvIK7zMzaBNFpKV3wAfW4GxQs0SBdpipiGvixyCPUx7Hw6nS6X4KRyDmYQ%2Bqhp1Bqi-9Yf2samtUlrU02TN3PR-BMexmxhEW5bGTWjattamG6ti%2BHDuOmxTsQb7fsZW72nu5zynWYjU0gOmGZxrB3oFoWRY0P7oIBoHK0ZMGIZWbmWwveHxopybpa4uX3IANUxEk7HgOx7DxmtCZq6Kebhk4zbYpHkq8dZNOt6Yzv0ytyod3Jnddgnqros9jfqn3yb9pKRXWVyQ6YAzYCwABxbBCuKxBSvK2PKvjmTE6w3nzRAtPKaeNLrY%2BwX5hJL5oEIEk6W77BBH4kh6X3JaKtTNCiYYkm2yUvDzYezJ1nauX5p4IWySwRUlnFxAjKyjhxPKn5clHVnyE5ifjyN2vvavVjDic5Gl%2BmuWAGlEEQVRPO8pYcmH3ACx5iTmvlPYaJsU4JQXlLJez1s5%2BUQoITQFcBpVwgLJPas84rz0bhbJeqM5aFAcJgHS9glpLGCogU4EwEClFQqAj2xMvak3hjgx%2B-sRQ4ignLH4Jl7AcDDhwCO3xCj8I0PESgAlYA8QoRocqICpI33khA%2B%2BiN06phxLLIY8tjqMg4HI6ICxvj2DPitfedgj6lgUfWBOtUk511UZLZ%2B3gcRW20SrWwYIFjUFOPAfABZYBlQcGvbgQtrHSXQTXZRydHHQOcV4HEwdtEGSZvgU4mAEFM00AIfI0xwlKJGu2KASx4Agi8WQfgmA1FN3KDiLO2jASlNBMA-65B0DAmhmAzBilsEqVwYvFxLdtEfy-jI-IORe74CEfjehiiukzx6awvp7D1FDBxCvbRncpmEC8YIHSelchr2EIgeAHj25cGLiVMqVinS-nmcwrBSyG4rJqZkHEr9Nm0jgnvJm0ByCmTsBoE5HpD7fCMtAQsdItTjyGJPRh08HmLPrg-G4T8A64jgdouQHxla0PyWAiwNQyghEEGdMqzg0L1CAA

dsyme commented:
@nojaf Here's another case of interest - records with XML doc comments on the fields.  Looks a bit odd to place that brace on the last line

https://fsprojects.github.io/fantomas-tools/#/fantomas/preview?data=N4KABGBEDGD2AmBTSAuKB6dYAijoBsBDAJwEsA7AczEPnlIBdTZzD8wBbRDgI0WIDOYFmAAKsAQzYBBOsUQCBAHWJLyDAJ4AHRGIlT8s%2BPMVgAvCrUQIwS%2BWsOwmLABUAFrtrGFy1fccQRiYCaJJkVHZ2ARDOYO660IwaUdEAwkmhDOGUkX7RTphxHmAAXqRaYHBIKQEAWuWZ2TVgAL6QADTgUAKIDExUAqhgANpdNmPWkAAkmjpDkBQMHRMQ0wBubACuyGij%2BQAMnfmQAJLkSOoAyqQlyEfRACwrYAC6Ey334wHTsztQi8tvlMNvhtkM9tEAIyfRyQACyhAAHgAZCiIZGIKgMNyA6IADn2zzeDg%2BE1AQN%2B8x4sFg%2BFxDnWWz%2BEICACYYQzLtxSHB8CxpAwAKLnADyADNUeQ7s8IGK2D0ie8YeTYTNtH9INTafTJsCmeCZWAAMwcyaXLSEaCIABCiDFsHkohIhC4DH4OocWW2ipJyueP3VVJpdNNjNBzMND1Nq3Nlptdod6NgAHd%2BNBCD0zmtYOmmCwPdYvYgfdZSQ4VQy1XM0JrgwWw2DdoaAKzRqCxq22%2B3yACqWh0xHTmfI2dzzHIBdl8uLAWJpb9FMDNa1If9evDBvyADY25AO-Hu4hUkRFKkWGFNtAGA7J2A5fgFbOlWS15Tl3XQ%2BvGyNDQB2Xf7l2iZwtwfDELe96Po4c4QGW1gVrqb5QCu9ZfhG%2BR4gBFqdgm8hnnyE5tpBM7Qc%2B5avkuyEfmuILfiyjgAJxYXG0him6xBnhwHCELeRYlrBC6qkhtbap%2BtHoVChyGnu2EHomXIcDytL5kR078a0gmVsJKFifqTb5JC0LSfurHsQpSkEbxxDek%2Bvovou1ZUaJNF6T%2BBnstJZwXAwIrkC4xAaAA6owOKqQ%2BJEODBGn2UJlEiauQLiZuUImsZsnSMQsCbOcuD4KQinsVZNmkXZ5EORqAK6Ru%2BlQlG0kIoiJxivE5CCuFlxuA6DDBfA2K3g8hK2fOMVaXFlUudVblQq29VImcYqkIiIoDoQ17EIKiJaME463s2g0lcNZWxY5CzqKhSU1QEkI7rNiIAEp4A68A9X1bYDepcFfMdFVnVVdGGpC-63Q9cDEPAABymy8Pw4onG6HCDG2kIfZpiFxRwmz4EweVSgAYg63F9PwLiBn9ElXZh0kg09cKY9jaL48QhOFbu0BuM6V78AA%2BsmpC9TiKMjWjJ3jYlrn0Q4kJMbdGXEIQGgisQqKSC9oWRvtkVkfBFEi79E3-fkrJSccDWy-LivKwwkPQ8QsPw4jAOC0do0nRjWOkDjiCM8zxOk-r5OOKyRnHGbCtK6Qki0%2B7nve6tLPSWzHPsTzfN9U72vlfMouqhdU1sh5JtIgAaky1oUPQVCq-1GvWFFn1gAhqxVj9Sxk8lbKpYXiK49lV7jmX5wUJQVdvTXEB16jTfCdnla5xL1isnVXfYLADAAOK9BtW0%2BOOI8tmPrxa19Lsajp-vt4HM0m3THtotafLQAA1tactP70Ai%2BZcLpHrSUOEYaxF07H2FqfaiYtJrzwgKyG6xxwaIGTJ7W0DBUyYhJjoXAC1yCMHHNIc4IEbYO3yIAoaAkhZTzimfcBBtoisiBscAA0ogRAWgmotTOF-LgkppRELUiQ6KzsQFZz1lQgODhWSUy7m1RSAg3B73yO9Ph9dG5QGbkGZyIiL5iOlsca4VB8CIAypQKGmJuqIB4HCBA3DojEIOqQgR5CTqUJzuLQ0RpjbREgNIPKlByA93IH3Fg1wfGrU2PIFwsAvImNWjtMKUFNalQzt9NRCVnEQNccHDxXj2KsCYGsJMVB8FgQwRQbB54IK8NsfwxJJ9knnRcfkI0BcPFRyYFw5ELoeDwEIMeCQQ84EILROU8KQCG461Aeo1J1CAhGk7h47AEdCA8H0VIiObhLgaHUEiIZcTa5H1GZnGsmJ4Dii4ZcTQ%2Bi6lpIaUvDxwpjkSkGazYg%2BAxSQBGcogMjiwGTNEdYI0V8PGMOYZE9QZwX6EH8WrHhwzFGTxUdpb5s96nRCNDAjx98IWPy4QIDKWVzhwMkIgeALTb5Si3ttMpbY%2BKwrIfCihiLdRz1cXQ9FJAgLyHmQIaAZBFI5KJT2LBLBcAEGdHmf%2B0Kdnjz2R81R74JlIquSiiRHizlkCvBYpA2yIq7JJF0F49wFgCFxgIUgQxAEtCAA

dsyme committed:
Update formatting.md

dsyme commented:
@nojaf OK I put the whole doc through the "preview" tools and apart from the cases above I think it's all looking good!

Please check over this commit: https://github.com/dotnet/docs/pull/25972/commits/6415eee20ed0a745aad802a5bba7bfeaac890ad6

dsyme committed:
Update formatting.md

dsyme commented:
@nojaf If you could check the above cases quickly that would be cool - if any of them are Fantomas bugs I can open issues.  I think maybe some are the changes we've discussed either not yet implemented or not in preview tools

nojaf commented:
Hey @dsyme, you are correct some stuff is not in the online tool yet (nor in a stable version).
Like
```fsharp
Target.create "Build" (fun ctx ->
    // code
    // here
    ())
```
is still open in https://github.com/fsprojects/fantomas/pull/1858. 
I'll get to that when time permits it.

```fsharp
let function1 arg1 arg2 arg3 arg4 = arg1 + arg2 + arg3 + arg4 + arg2 + arg3 + arg4 + arg2 + arg3 + arg4
```
Fantomas currently treats almost all infix operators the same. There is nothing in place for arithmetic operators and in general infix operators are quite challenging to get right. It is a bit on an ongoing dragon in the codebase. I'm open to some good heuristics there.

```fsharp
let rainbow2 =
    { rainbow with
        Boss = "Jeffrey"
        Lackeys = [ "Zippy"; "George"; "Bungle" ] }
```

Created https://github.com/fsprojects/fantomas/issues/1876

```fsharp
let makeStreamReader x = new System.IO.StreamReader(path=x)
```
we just format the shape of the argument AST here as is.
```fsharp
Paren
                       (App
                          (NonAtomic, false,
                           App
                             (NonAtomic, true, Ident op_Equality, Ident path,
                              tmp.fsx (1,52--1,57) IsSynthetic=false), Ident x,
                           tmp.fsx (1,52--1,58) IsSynthetic=false),
                        tmp.fsx (1,51--1,52) IsSynthetic=false,
                        Some tmp.fsx (1,58--1,59) IsSynthetic=false,
                        tmp.fsx (1,51--1,59) IsSynthetic=false),
                     tmp.fsx (1,25--1,59) IsSynthetic=false),
```
we should do something different when it is in the context of SynExpr.New.
Created https://github.com/fsprojects/fantomas/issues/1877

```fsharp
// Declaring additional members on PostalAddress
type PostalAddress =
    {
        /// The address
        Address: string

        /// The city
        City: string

        /// The zip code
        Zip: string
    }
```
to preserve the original style here, you are looking for the [fsharp_multiline_block_brackets_on_same_column](https://github.com/fsprojects/fantomas/blob/master/docs/Documentation.md#fsharp_multiline_block_brackets_on_same_column) setting perhaps?

[Example](https://fsprojects.github.io/fantomas-tools/#/fantomas/preview?data=N4KABGBEDGD2AmBTSAuKB6dYAijoBsBDAJwEsA7AczEPnlIBdTZzD8wBbRDgI0WIDOYFmAAKsAQzYBBOsUQCBAHWJLyDAJ4AHRGIlT8s%2BPMVgAvCrUQIwS%2BWsOwmLABUAFrtrGFy1fccQRiYCaJJkVHZ2ARDOYO660IwaUdEAwkmhDOGUkX7RTphxHmAAXqRaYHBIKQEAWuWZ2TVgAL6QADTgUAKIDExUAqhgANpdNmPWkAAkmjpDkBQMHRMQ0wBubACuyGij%2BQAMnfmQAJLkSOoAyqQlyEfRACwrYAC6Ey334wHTsztQi8tvlMNvhtkM9tEAIyfRyQACyhAAHgAZCiIZGIKgMNyA6IADn2zzeDg%2BE1AQN%2B8x4sFg%2BFxDnWWz%2BEICACYYQzLtxSHB8CxpAwAKLnADyADNUeQ7s8IGK2D0ie8YeTYTNtH9INTafTJsCmeCZWAAMwcyaXLSEaCIABCiDFsHkohIhC4DH4OocWW2ipJyueP3VVJpdNNjNBzMND1Nq3Nlptdod6NgAHd%2BNBCD0zmtYOmmCwPdYvYgfdZSQ4VQy1XM0JrgwWw2DdoaAKzRqCxq22%2B3yACqWh0xHTmfI2dzzHIBdl8uLAWJpb9FMDNa1If9evDBvyADY25AO-Hu4hUkRFKkWGFNtAGA7J2A5fgFbOlWS15Tl3XQ%2BvGyNDQB2Xf7l2iZwtwfDELe96Po4c4QGW1gVrqb5QCu9ZfhG%2BR4gBFqdgm8hnnyE5tpBM7Qc%2B5avkuyEfmuILfiyjgAJxYXG0him6xBnhwHCELeRYlrBC6qkhtbap%2BtHoVChyGnu2EHomXIcDytL5kR078a0gmVsJKFifqTb5JC0LSfurHsQpSkEbxxDek%2Bvovou1ZUaJNF6T%2BBnstJZwXAwIrkC4xAaAA6owOKqQ%2BJEODBGn2UJlEiauQLiZuUImsZsnSMQsCbOcuD4KQinsVZNmkXZ5EORqAK6Ru%2BlQlG0kIoiJxivE5CCuFlxuA6DDBfA2K3g8hK2fOMVaXFlUudVblQq29VImcYqkIiIoDoQ17EIKiJaME463s2g0lcNZWxY5CzqKhSU1QEkI7rNiIAEp4A68A9X1bYDepcFfMdFVnVVdGGpC-63Q9cDEPAABymy8Pw4onG6HCDG2kIfZpiFxRwmz4EweVSgAYg63F9PwLiBn9ElXZh0kg09cKY9jaL48QhOFbu0BuM6V78AA%2BsmpC9TiKMjWjJ3jYlrn0Q4kJMbdGXEIQGgisQqKSC9oWRvtkVkfBFEi79E3-fkrJSccDWy-LivKwwkPQ8QsPw4jAOC0do0nRjWOkDjiCM8zxOk-r5OOKyRnHGbCtK6Qki0%2B7nve6tLPSWzHPsTzfN9U72vlfMouqhdU1sh5JtIgAaky1oUPQVCq-1GvWFFn1gAhqxVj9Sxk8lbKpYXiK49lV7jmX5wUJQVdvTXEB16jTfCdnla5xL1isnVXfYLADAAOK9BtW0%2BOOI8tmPrxa19Lsajp-vt4HM0m3THtotafLQAA1tactP70Ai%2BZcLpHrSUOEYafEhoCSFlPOKZ8xaTXnhAVkN1jjg0QMmT2toGCpkxCTHQuAFrkEYOOaQ5wQI2wdvkYi6dj7C1PtRCBBtoisiBscAA0ogRAWgmotTOF-LgkppTELUkA6KztyFZz1lQgODhWSUy7m1RSAg3B73yO9Ph9dG5QGbkGZyIiL5iOlsca4VB8CIAypQKGmJuqIB4HCBA3DogkMUZPFR2lKE53FoaI0xtoiQGkHlSg5Ae7kD7iwa43jVqbHkC4WAXkTGrR2mFKCmtSoZ2%2BmohKTjIEuODu4zx7FWBMDWEmKgBCwKYIoDg88EFeEHWAQI0BJ1wEpOoQEI0Bd3FRyYFw5ELoeDwEIMeCQQ94GILRGU8KpCG46woeoupojrBGk7u47AEdCA8H0VIiObhLgaHUEiIZsTa5H1GZnGsmJ4Dii4ZcTQ%2BjzrOPyEaJe7jhTHIlIM1mxB8BikgCM5RAYamONnlc6IRor7uMYcwiJ6gzgv0IH4tWPDhm2JAfYsBPzdRzxcbA9x99IWPy4QIDKWVzjwMkIgeALTb5Si3ttUpbZAEVP4Qkk%2BSTLmpOuXQ9FJAgLyHmQIaAZBFLZKJT2bBLBcAEGdHmf%2BMKdnjz2Z81R74Jm-KZf8iR7izlkCvBYpA2yIq7JJF0F49wFgCFxgIUgQwSEtCAA).

dsyme commented:
@nojaf Regarding records declarations (with/without XML doc comments):

> to preserve the original style here, you are looking for the fsharp_multiline_block_brackets_on_same_column setting perhaps?

This is tricky.  Here's my thinking

1. My natural instinct (not necessarily good) is to have records without XML doc be formatted compactly (record definition braces on same column = false) while those with XML doc be formatted widely (record definition braces on same column = true). This is roughly what I implemented for signature file generation in https://github.com/dotnet/fsharp/pull/12072/  

2. However, this means a discontinuous approach that people may find confusing. And also it only applies to record definitions, and not "multiline_block_brackets" more generally, because that's the only place in the syntax where XML doc comments interact with block brackets.

3. Now there's a strong argument that the "destination" of code should be XML-doc commented.  So if I really have to choose one or the other for default formatting then I should be choosing wide formatting (record definition braces on same column = true)

Anyway, on summary on revising the guide I kind of feel that record definitions should get special treatment to support a nice look when XML doc commented.   But I'm aware that this is a special case.  Would be interested in your thoughts.

nojaf commented:
Hmm, a bit tricky indeed. Although as I understand it there is a clear rule when to always use the wide record formatting:
- 1. record type (in implementation or signature file)
- 2. all fields have XML comments

This is a relatively easy thing to check in the syntax tree.
From a technical point of view, I think we can have this.
Although it will be raised multiple times on Fantomas side as inconsistent no doubt 🙈.
I would be ok with having this as there is clear reasoning behind it and it is scoped to a limited number of cases.

Slightly related I've raised https://github.com/fsprojects/fantomas/issues/1878.

dsyme commented:
@nojaf Added https://github.com/fsprojects/fantomas/issues/1879 where we can discuss

nojaf commented:
I've updated the tool to use our `4.6` branch which has the [new lambda style](https://fsprojects.github.io/fantomas-tools/#/fantomas/preview?data=N4KABGBEDGD2AmBTSAuKAVAhgJwOaIBcA6abRTAxMAHUgCEBXASwBt5awAKAMwYDsw0AgA8wAWgB81bNQEQwAegWCEiabPmLlAC0Rl1ciJwCUxyABpwUAM6ECTPrmuowAbSsRQmiJAAkBAE8AB2Q0SAcCCw95PwA3TBYGULdozQAGS28YgEk%2BJD4CAGUmAC9kTKywABZUsABdaIBfCs9av0CQl3CCqKy4hKSXd0qwAEYW70gAWUxhABkHRDnERwJtXsrRgCY02obNZuivSf9g5MgAI1hYFg3NfsTk4cqtifvCxABbJjgWWD4AIIEACieQA8twFnxyrUINwErY9k0Jsd7qdOmErjc7jFfPFHkNYWAAMxvGKFIKYaCIOiIbiwMgABRwmE%2BhD0OM0BGwSSRBxRbXR5yxtzJD0GaGeWSqZJ8FKpNLpDKWsAA7npoJhbLlYrBNfZ-pz5NzeVl9vJDppUbiOsLrqLBfiJSkRgBWWVQeXU2n0sgAVSCIWwmu1fF1%2BqYho98JYiLNyKOgttXRFRvFTyJADYPZAvYrfYgAMIsLXWQv-awmoQMo1whGIPkWgV9IUp%2B1pvEDDMjADsObzPuVUy%2BFw50frjYglvk1p8rcx7bFnYJkqJAA5%2B5TvUqyOW-nxa2AY3HvOap82TsmF9il07u5UAJybhUA7iUbDlz6fTCHk0N%2BP8omLZXlAqa3l2hIjKMGRErmW75sqHzfL8UZEse-6ngmVpJmcbY3o6EGrlB4ywXmr7vkhPw3KhIx-pOYDTq0wG4deDotnekGbK8sG5PkBBgnw6DYAEADqTBrIe6H0YxYCzlA86gYuBEri6mykqR8EAtgsD8PAAAiiAsEw3zvr%2BPIYZoZ4MReaIgd0kTgSpUreKMMqwTMwjZNw6C6HwwKxoghTaAyBBifAEkelUuwAU2QGXixUARB2HFEZs7rubMuTcEwwhgkGFAMsCwhBGQ1jWJGB4eq60WYYB2HMRiiU9I5zrOZoozZhlwgAEqIHA2DwGFEVElF0k2TaCX2clhGqVkox9l1vX9fAAByDCfKO2AQtklCfM4HqjGNcW2ZNnwMCw9hGdCABiDLfgQ77oLhLX3nNG6wUtDLwFM52XYst3YPdpk5tA2gskIegAPqqkw4XrEd9XxY1U0vZxc1Pl1WnYJgARgtgCyVkN6yRTVllYTOOHI0lqOpVkOw5h5WM43jBMEGtG16Ntu37USh0xeex0TcjZ0Xaw-13RQj3PcprVElsJEjJATO4-jTCVj9otXYgANA2OsGg%2BD77Q7DEkIxTDXnNTMuvd4WzcYrHkAGpdnQDjwA4uBE4eo389ZgtznZVvsTNbXyFs6kO7M138EIFWu3kHteyTZtMUjlvNdbaO225kfCHpsAEAA4oQRUlYgZUVUnRLVSnsmU3a%2BHB05cvpQ7v1i9CdB-NAADWdDY73hDWAJhSskWNzrZVaETr7Mlye0k1gZntO251isrYgqpa7SBDqisT0hAZ2V8OJFUAnkw4c9gPMjFJs-jQHi9KU3ssjFsC2KwA0ogiBBF5PkrFyKPNkUIYS3xnrVWKiMTpUwzi-G2mgtjvVzv5b41htBVxGD7SBAtoFCwbmxE4KVZq2wxorYojgWCIC0rgdaKxQqIAuFMVQkkIFkzqubNOeFCFomIaHCAxIYKKwBEZXAfBo58Fjv8YoYiKAMDIOgWAvF6EUAqqwgKtd54KUuM-IhIciTEgVpUJWF09B8FUbEFUjhL6bSPg4U%2BFZ1EnnYVAzhMCCHTWbiMYk9tjEa3sKAuYrILjwEwMWWA5VHAby3osJxFl5BWTnvXbhnjX6VGJBHYxek1aYAuFQ1BattCFACAUWYcTNHJLCCseAEJQGFECFQ1JCD5DEhzsY0ENTISxJBtgFg3BIAVItikmmJDNDElbsY7%2Bv9lEFFyP3TAkjibTw0fff28k7JL3gVnMZa9jFdwWT3UB1gtI6TyBvSsiBvrty1qXUq5VHEejoqsvBj9kabL0V49JH89k4EHGQbJ1hSDGQcBQS5foT7-AMtAEs2MDRT3ASsnBfsXnrKfo3D5aSsjEmQcY%2Bp2AfgEGYUgcps8rB1AqOEaw11youCko0IAA).

luisquintanilla commented on docs/fsharp/style-guide/formatting.md:
```diff
@@ -116,657 +62,512 @@ This is sometimes called “vanity alignment” or “vanity indentation”. The
-type MyRecord = { IntVal: int; StringVal: string }
+let poemBuilder = StringBuilder()
+poemBuilder.AppendLine(
+    """
+The last train is nearly due
+The Underground is closing soon
+And in the dark, deserted station
+Restless in anticipation
+A man waits in the shadows
+    """
```
Was this the intended indentation?

```suggestion
    """
    The last train is nearly due
    The Underground is closing soon
    And in the dark, deserted station
    Restless in anticipation
    A man waits in the shadows
    """
```

dsyme replied:
No, this is a triple-quote string, it's intentional to left-indent

luisquintanilla commented on docs/fsharp/style-guide/formatting.md:
```diff
@@ -116,657 +62,512 @@ This is sometimes called “vanity alignment” or “vanity indentation”. The
+let performAction =
+    complexExpression a b && env.IsDevelopment()
+    || secondLongerExpression
+        aVeryLongparameterNameOne
+        aVeryLongparameterNameTwo
+        aVeryLongparameterNameThree
```
Is there an extra space in these params?

dsyme replied:
No, I don't think so - they are parameters to a function

luisquintanilla commented on docs/fsharp/style-guide/formatting.md:
```diff
@@ -858,90 +657,88 @@ let daysOfWeek' includeWeekend =
+// OK
+let rainbow2 =
+    { rainbow with
+        Boss = "Jeffrey"
+        Lackeys = [ "Zippy"; "George"; "Bungle" ] }
```
Is there an extra / missing space in these labels?

dsyme replied:
I'm not sure what's suggested here?

luisquintanilla commented on docs/fsharp/style-guide/formatting.md:
````diff
@@ -1040,210 +852,408 @@ with
 ```
 
-## Formatting function parameter application
+### Formatting named arguments
````
Should this be an H2 instead of an H3?

dsyme replied:
Went through and fixed all these, some H2 --> H3 and some H3 --> H2

luisquintanilla commented on docs/fsharp/style-guide/formatting.md:
````diff
@@ -1040,210 +852,408 @@ with
 ```
 
-When pipelines are concerned, the same is typically also true, where a curried function is applied as an argument on the same line:
+### Formatting object expressions
````
Should this be an H2 instead of an H3?

luisquintanilla commented on docs/fsharp/style-guide/formatting.md:
````diff
@@ -1040,210 +852,408 @@ with
+let makeStreamReader x = new System.IO.StreamReader(path = x)
+```
+
+### Formatting mutation expressions
````
Should this be an H2 instead of an H3?

luisquintanilla commented on docs/fsharp/style-guide/formatting.md:
```diff
@@ -1040,210 +852,408 @@ with
-Unless there are many leading or multiline arguments before the lambda. In that case, indent all arguments with one scope.
+For any custom operator that starts with `*` and that has more than one character, you need to add a white space to the beginning of the definition to avoid a compiler ambiguity. Because of this, we recommend that you simply surround the definitions of all operators with a single white-space character.
+
+## Formatting record declarations
```
Since this is a declaration, should it be an H3 since it would fit under the broader "Formating declarations" section?

luisquintanilla commented on docs/fsharp/style-guide/formatting.md:
````diff
@@ -1040,210 +852,408 @@ with
 ```
 
-When the function take a single multiline tuple argument, the same rules for [Formatting constructors, static members, and member invocations](#formatting-constructors-static-members-and-member-invocations) apply.
+## Formatting discriminated union declarations
````
Should this be an H3 since it falls under broader "Formatting declarations" section?

luisquintanilla commented on docs/fsharp/style-guide/formatting.md:
```diff
@@ -1040,210 +852,408 @@ with
-### Formatting modules
+Avoid placing the attribute on the same line as the value.
+
+## Formatting module declarations
```
Should this be an H3 since it falls under broader "Formatting declarations" section?

luisquintanilla commented on docs/fsharp/style-guide/formatting.md:
```diff
@@ -1263,100 +1273,203 @@ module A2 =
 
-### Formatting white space in expressions
+## Formatting types and type annotations
+
```
Maybe add a brief intro sentence here for the section?

dsyme replied:
Done, thanks

luisquintanilla commented on docs/fsharp/style-guide/formatting.md:
````diff
@@ -1263,100 +1273,203 @@ module A2 =
 ```
 
-If the expression is long, use newlines and indent one scope, rather than indenting to the bracket.
+## Formatting types in signatures
````
Should this be H3 since it falls under broader "Formatting Types and Type Annotations" section?

luisquintanilla reviewed (commented):
Thanks all for making these updates. Looks great. Just a few minor formatting issues 😉.

Can you also update the `ms.date` value in the metadata to reflect this document has been recently updated.

dsyme commented on docs/fsharp/style-guide/formatting.md:
```diff
@@ -116,657 +62,512 @@ This is sometimes called “vanity alignment” or “vanity indentation”. The
-type C() =
-    member _.Property: int = 1
+let (x, y) = z
+let x, y = z
```
From the line before: "It is commonly accepted to omit parentheses in pattern matching of tuples"

dsyme committed:
Update formatting.md

dsyme committed:
Update formatting.md

dsyme commented:
@luisquintanilla This is ready, thank you!!

> Thanks all for making these updates. Looks great. Just a few minor formatting issues 😉.

Thanks, all done

> Can you also update the ms.date value in the metadata to reflect this document has been recently updated.

Updated

luisquintanilla reviewed (approved):
Looks good to me. Ready for merge.

dsyme commented:
@luisquintanilla Thanks!
